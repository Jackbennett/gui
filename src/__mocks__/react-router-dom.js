import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
