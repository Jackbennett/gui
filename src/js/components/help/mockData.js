export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
