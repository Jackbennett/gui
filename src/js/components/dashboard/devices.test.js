import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import Devices from './devices';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('Devices Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      deployments: {
        deploymentDeviceLimit: 5000
      },
      devices: {
        byId: {},
        byStatus: {
          accepted: {
            total: 0,
            deviceIds: []
          },
          active: {
            total: 0
          },
          inactive: {
            total: 0
          },
          pending: {
            total: 0
          }
        }
      },
      users: {
        onboarding: { complete: false },
        showHelptips: true
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Devices getAllDevicesByStatus={jest.fn()} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
