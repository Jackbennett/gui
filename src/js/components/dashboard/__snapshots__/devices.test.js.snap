// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Devices Component renders correctly 1`] = `
<div>
  <h4
    className="dashboard-header"
  >
    <span>
      Devices
    </span>
  </h4>
  <div
    style={
      Object {
        "marginBottom": 30,
      }
    }
  >
    <div
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiFab-root MuiFab-primary"
        href="/devices/pending"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        style={
          Object {
            "left": "15px",
            "position": "absolute",
            "top": "-28px",
            "zIndex": "1",
          }
        }
        tabIndex={0}
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
      <div
        className="MuiPaper-root widget  MuiPaper-elevation2 MuiPaper-rounded"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <div
            className="widgetMainContent align-right"
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
                "flexDirection": "column",
              }
            }
          >
            <div
              className="header"
            >
              Pending devices
            </div>
            <div
              className="counter"
            >
              1
            </div>
          </div>
          <span
            className="link"
          >
            View details
          </span>
        </div>
      </div>
    </div>
    <div
      className="notActive widget "
      onClick={[Function]}
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <div
          className="widgetHeader"
          style={
            Object {
              "borderBottomStyle": "solid",
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              style={
                Object {
                  "color": "#009e73",
                  "height": "14px",
                  "marginRight": "10px",
                  "width": "14px",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <div
                className="hint"
              >
                All devices online
              </div>
              <div
                className="tiny"
              >
                Active in past 24h
              </div>
            </div>
          </div>
        </div>
        <div
          className="widgetMainContent align-right"
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <div
            className="header"
          >
            Accepted devices
          </div>
          <div
            className="counter"
          >
            2
          </div>
        </div>
        <span
          className="link"
        />
      </div>
    </div>
    <div
      className="onboard widget"
      onClick={[Function]}
    >
      <div>
        <p
          className="muted"
        >
          Learn how to connect a device
        </p>
      </div>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        href="/devices"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          Connect a device
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
